### ISSUE TITLE
ec2-tag-compliance documentation needs adjusting

### ISSUE THREAD

**Issue by amphibithen** [capitalone/cloud-custodian#856]
i was using the recommended lambda configuration for [ec2-tag-compliance](http://www.capitalone.io/cloud-custodian/docs/policy/lambda.html) however instead of action:STOP i used action:TERMINATE.

```
policies:

  - name: ec2-tag-compliance
    resource: ec2
    mode:
      type: cloudtrail
      events:
       - RunInstances
    filters:
      - or:
        - tag:required1: absent
        - tag:required2: absent
    actions:
      - type: terminate
        force: true
```

this was working for a week, but then today it decided to start terminating one of my ec2s attached to asg (that had the required tags), and did it many times in a row.  looking at cloudtrail, i realized event=RunInstances and event=CreateTags had a 30 second delay.  in the past this happened within 1 second.  Later in the day it went to 1+ minute.

So ASG would spin a new EC2, lambda would start to terminate it, then ASG would apply the tags.

I believe the ec2-tag-compliance should instead use the event=CreateTags, instead of RunInstances.  So it would look something like:
```
policies:

  - name: ec2-tag-compliance
    resource: ec2
    mode:
      type: cloudtrail
      events:
        - source: ec2.amazonaws.com
          event: CreateTags
          ids: "requestParameters.resourcesSet.items[].resourceId"
      role: arn:aws:iam::
    filters:
      - or:
        - tag:required1: absent
        - tag:required2: absent
    actions:
      - type: terminate
        force: true
```

This is also a more logical place to put the event i believe, as you are checking for a tag, so better to wait until the tag is created before checking it.

**Comment by kapilt** [comment]
So since ec2 runinstances doesn't support tags, doing this policy in lambda and avoiding race conditions its probably best as a config rule. CreateTags, won't handle someone doing RemoveTags.

**Comment by whit537** [comment]
PR in https://github.com/capitalone/cloud-custodian/pull/965.

**Comment by whit537** [comment]
Closeable with #965, or do we want to actually add Lambda instructions to existing tag compliance docs as well?

**Comment by whit537** [comment]
Considering closed with #965.
